src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java21/InputStringTemplateBasic.java

```diff
@@ -0,0 +1,49 @@
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.grammar.java21;
+
+public class InputStringTemplateBasic {
+    int x = 10;
+    int y = 20;
+
+    // most basic example, empty string template expression
+    String result1 = STR."";
```

Comment:
```
        |--VARIABLE_DEF -> VARIABLE_DEF [9:4]
        |   |--MODIFIERS -> MODIFIERS [9:4]
        |   |--TYPE -> TYPE [9:4]
        |   |   `--IDENT -> String [9:4]
        |   |--IDENT -> result1 [9:11]
        |   |--ASSIGN -> = [9:19]
        |   |   `--EXPR -> EXPR [9:24]
        |   |       `--DOT -> . [9:24]
        |   |           |--IDENT -> STR [9:21]
        |   |           `--STRING_LITERAL -> "" [9:25]
```

Reply:
@rnveach @romani I am not sure about this one, should we do the above, or like this:
```
        |   |       `--DOT -> . [15:24]
        |   |           |--IDENT -> STR [15:21]
        |   |           `--STRING_TEMPLATE_BEGIN -> "
        |   |               |--STRING_TEMPLATE_CONTENT ->
        |   |               `--STRING_TEMPLATE_END -> "

```

Reply:
I suppose it is still a template, I will update the code.

Reply:
```
        |   |--ASSIGN -> = [9:19]
        |   |   `--EXPR -> EXPR [9:24]
        |   |       `--DOT -> . [9:24]
        |   |           |--IDENT -> STR [9:21]
        |   |           `--STRING_LITERAL -> "" [9:25]
```
is consistent with what we do now.

```
        |   |       `--DOT -> . [15:24]
        |   |           |--IDENT -> STR [15:21]
        |   |           `--STRING_TEMPLATE_BEGIN -> "
        |   |               |--STRING_TEMPLATE_CONTENT ->
        |   |               `--STRING_TEMPLATE_END -> "
```
is to clear pointer it is String Template syntax.

if we reuse old tree, we might immediately need to update all our Checks to exclude this from consideration. and it will be hard to find by AST that it is String Template.

I  vote for STRING_TEMPLATE_BEGIN approach.